app/services/measurements_creator.rb

```diff
@@ -19,12 +21,19 @@ def call(stream, measurements_attributes)
     stream.build_measurements!(measurements_attributes)
     stream.after_measurements_created
 
-    return if stream.session.type == 'FixedSession'
+    return if stream.fixed?
     streams_repository.calc_bounding_box!(stream)
     streams_repository.calc_average_value!(stream)
   end
 
   private
 
   attr_reader :streams_repository
+
+  def self.add_arrival_time(measurements_attributes)
+    measurements_attributes.map do |measurement_attributes|
+      measurement_attributes[:arrival_utc_time] = Time.current.utc
```

Comment:
Does `Time.current` give you non utc? If that's the case I think the Rails config is wrong. Maybe it would be better to fix the config and drop the `.utc`?